docs/source/bytes.rst

```diff
@@ -0,0 +1,73 @@
+Internal Data Ingestion
+=======================
+
+Dask contains internal tools for extensible data ingestion in the
+``dask.bytes`` package.  *These functions are developer-focused rather than for
+direct consumption by users.  These functions power user facing functions like
+``dd.read_csv`` and ``db.read_text`` which are probably more useful for most
+users.*
+
+.. currentmodule:: dask.bytes
+
+.. autosummary::
+   read_bytes
+   open_files
+   open_text_files
+
+These functions are extensible in their output formats (bytes, file objects),
+their input locations (file system, S3, HDFS), line delimiters, and compression
+
+These functions provide data as ``dask.delayed`` objects.  These objects either
+point to blocks of bytes (``read_bytes``) or open file objects (``open_files``,
+``open_text_files``).  They can handle different compressions by prepending
+protocols like ``s3://`` or ``hdfs://``.  They handle compressions listed in the
+``dask.bytes.compression`` module.
+
+These functions are not used for all data sources.  Some data sources like HDF5
+are quite particular and receive custom treatment.
+
+
+Delimiters
+----------
+
+The ``read_bytes`` function takes a path (or globstring of paths) and produces
+a sample of the first file and a list of delayed objects for each of the other
+files.  If passed a delimiter such as ``delimiter=b'\n'`` it will ensure that
+the blocks of bytes start and end on this delimiter.  This allows other
```

Comment:
May need to clarify for inclusivity. Currently this reads to me as returning "\nsometext\n".